contracts/core/KernelGateway.sol

```diff
@@ -166,6 +180,84 @@ contract KernelGateway is KernelGatewayInterface{
         );
     }
 
+    /**
+     * @notice Prove origin core account and update the latest storage root of
+     *         origin core. This function will validate the proof against the
+     *         state root from the OriginBlockStore
+     *
+     * @param _accountRlp RLP encoded account data.
+     * @param _accountBranchRlp RLP encoded path from root node to leaf node to
+     *                          prove account in merkle tree.
+     * @param _originBlockHeight The block height on origin where the opening
+     *                           took place. Must be a height that is finalised
+     *                           in the OriginBlockStore.
+     *
+     * @return success_ `true` if the proof is successful.
+     */
+    function proveOriginCore(
```

Comment:
This method is a potential duplicate of `proveGateway` method in `GatewayBase.sol`
[proveGateway](
https://github.com/OpenSTFoundation/mosaic-contracts/blob/617047f955b7a3bf88df7c5599650046ab282d99/contracts/gateway/GatewayBase.sol#L230)

I think logic to proven an account(even storage root `mapping`) should be extracted out in a contract. 

what do you think @schemar  @deepesh-kn  ?

Reply:
Good point. Let us create a ticket for this.

Reply:
I also think we should create a ticket for this 👍

Reply:
Created a ticket for this.
https://github.com/OpenSTFoundation/mosaic-contracts/issues/478